wemake_python_styleguide/visitors/ast/keywords.py

```diff
@@ -73,6 +74,25 @@ def _check_last_return_in_function(self, node: ast.Return) -> None:
         if node is parent.body[-1] and node.value is None:
             self.add_violation(InconsistentReturnViolation(node))
 
+    def _check_variables_used_only_in_return(
+        self,
+        node: AnyFunctionDef,
+    ) -> None:
+        variable_name = ''
+        return_variable_id = ''
+        if len(node.body) > 1:
```

Comment:
After writing new tests you will find bugs in this code. It should be rewritten.